tests/playbooks/tests_route_next_hop_ethernet.yml

```diff
@@ -0,0 +1,188 @@
+            that:
+              - route_table_main_ipv6.stdout is search("2001:db6::4 via
+                2001:db8::1 dev ethtest0\s+(proto static )?metric 2")
+            msg: "the route table main does not contain the specified IPv6
+              route"
+        - name: Get the interface1 MAC address
+          command: cat /sys/class/net/{{ interface1 }}/address
+          register: interface1_mac
+          ignore_errors: true  # noqa ignore-errors
+          changed_when: false
+          when: network_provider == "initscripts"
+        - name: Configure the network with only the MAC address specified
+          import_role:
+            name: linux-system-roles.network
+          vars:
+            network_connections:
+              - name: "{{ interface1 }}"
+                mac: "{{ interface1_mac.stdout }}"
+                type: ethernet
+                autoconnect: false
+                ip:
+                  dhcp4: false
+                  address:
+                    - 198.51.100.4/24
+                    - 2001:db8::6/32
+                  route:
+                    - network: 198.58.10.64
+                      prefix: 26
+                      gateway: 198.51.100.102
+                      metric: 4
+          when: network_provider == "initscripts"
+
+        - name: Assert that the warning about specifying the route without
+            the output device is logged
+          assert:
+            that:
+              - __network_connections_result.stderr is search("The connection
+                .* does not specify an interface name. Therefore, the route .*
+                will be configured without the output device and the kernel
+                will choose it automatically which might result in an unwanted
+                device being used. To avoid this, specify `interface_name` in
+                the connection appropriately.")
+            msg: The warning about specifying the route without the output
+              device is not logged
+          when: network_provider == "initscripts"
+      always:
+        - name: Remove test configuration
+          block:
+            - name: Bring down test devices and profiles
+              include_role:
+                name: linux-system-roles.network
+              vars:
+                network_connections:
+                  - name: "{{ interface0 }}"
+                    persistent_state: absent
+                    state: down
+                  - name: "{{ interface1 }}"
+                    persistent_state: absent
+                    state: down
+              ignore_errors: true
+            - name: Assert interface0 profile is absent
+              include_tasks: tasks/assert_profile_absent.yml
+              vars:
+                profile: "{{ interface0 }}"
+            - name: Assert interface1 profile is absent
+              include_tasks: tasks/assert_profile_absent.yml
+              vars:
+                profile: "{{ interface1 }}"
```

Comment:
Seems that this could be a loop.

Reply:
I used the loop to simplify the testing code.